### ISSUE TITLE
Severe pulsing when using multiple dimmers.

### ISSUE THREAD

**Issue by garrettdowd** [RobotDynOfficial/RBDDimmer#6]
I am using this library with a Arduino Uno (AVR ATMEGA328P). I have come across two main problems.

1) The frequency of dimming drops (pulses are more noticeable) as more dimmers are used (single zero-cross with multiple output/"PWM" pins). Beyond 3 dimmers the timing is not consistent enough to use with dimmable LEDS. The pulses are clearly visible on incandescent bulbs.

2) The timing is unreliable. Adding dimmers seems to alter the timing of the dimming and thus the brightness. So a power level of 50% with 2 dimmers is visibly different than a power level of 50% with 6 dimmers.

I need to be able to use 5 dimmers for my project and at this point the library seems unusable for this case.

Is this somehow related the the interrupt capabilities of the arduino? Can a faster timer interrupt be set?

**Comment by sircommitsalot** [comment]
@garrettdowd I'm encountering this issue as well. Did you come up with a solution?

**Comment by garrettdowd** [comment]
@jkprow Yeah, I used a different library. https://github.com/circuitar/Dimmer

**Comment by sircommitsalot** [comment]
That library works great, thanks for the tip.

**Comment by HFsi** [comment]
Does anyone tested it in ESP8266?

**Comment by fabianoriccardi** [comment]
It seems that Circutar's library doesn't work on ESP8266 (neither ESP32), it's only AVR compatible (it depends on specific hardware timer). You may try [this alternative](https://github.com/fabiuz7/Dimmable-Light-Arduino)
